### PR TITLE
Add comments

### PR DESCRIPTION
**Pull request by DanieleVBella**
Work as expected with users; comments are owned by users, and can only be edited by their owners. 
Can be viewed on their respective post page.

There will be some conflicts with pull request #6. **Merge this one second**.

The conflict at `db/schema`, should be resolved in the usual manner by dropping and recreating the database. For all other conflicts, the elements of both branches should be included i.e. nothing should be deleted.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to current_post, notice: 'Comment was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def edit; end
```
Comment (vishthemenon) [inline app/controllers/comments_controller.rb]: This is not needed. The updates will be done through an ajax request (PUT /comment/:id)

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+class CommentsController < ApplicationController
+  before_action :set_comment, only: [:edit, :update, :destroy]
+  before_action :check_permission, only: [:edit, :destroy]
+
+  def new
```
Comment (vishthemenon) [inline app/controllers/comments_controller.rb]: This is not needed. The updates will be done through an ajax request (POST /comment)

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  def comment_params
+    params.require(:comment).permit(:body)
+  end
+
+  def current_post 
+    Post.find(session[:last_post_id])
+  end
+
+  def check_permission
```
Comment (vishthemenon) [inline app/controllers/comments_controller.rb]: This method could be moved to ApplicationsController or an authentication helper as it is being reused in multiple controllers

File: config/routes.rb [hunk 4]
```diff
@@ -2,6 +2,8 @@
   devise_for :admin_users, ActiveAdmin::Devise.config
   ActiveAdmin.routes(self)
   devise_for :users
+  resources :comments, only: [:new, :create, :edit, :update, :destroy]
+  get 'comments/destroy'
```
Comment (vishthemenon) [inline config/routes.rb]: This is not an idiomatic REST resource. GET paths should not modify/create/destroy resources

File: app/views/comments/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<%= form_with(model: comment, local: true) do |form| %>
+  <% if comment.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(comment.errors.count, "error") %> prohibited this comment from being saved:</h2>
```
Comment (vishthemenon) [inline app/views/comments/_form.html.erb]: All the views are unused

File: app/models/post.rb [hunk 6]
```diff
@@ -1,5 +1,6 @@
 class Post < ApplicationRecord
   belongs_to :subject
+  has_many :comments, as: :commentable
```
Comment (vishthemenon) [inline app/models/post.rb]: Whats with the `commentable`

Comment (DanieleVBella) [reply app/models/post.rb]: It is required for the polymorphic association to work.

**Pull request merged**